dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -332,175 +406,168 @@ class _WidgetBuildRecorderHostState extends State<_WidgetBuildRecorderHost> {
-    if (profile.drawFrameDurationNoise > _kNoiseThreshold) {
-      // If the benchmark has run long enough, stop it, even if it's noisy under
-      // the assumption that this benchmark is always noisy and there's nothing
-      // we can do about it.
-      if (_frames.length > _kMaxSampleCount) {
-        print(
-          'WARNING: Benchmark noise did not converge below ${_kNoiseThreshold * 100}%. '
-          'Stopping because it reached the maximum number of samples $_kMaxSampleCount. '
-          'Noise level is ${profile.drawFrameDurationNoise * 100}%.',
-        );
-        return false;
-      }
+  double get noise => standardDeviation / average;
 
-      // Keep running.
-      return true;
+  void add(num value) {
+    count++;
+    _values.add(value);
+    // Don't let the list grow beyond [_kMeasuredSampleCount].
+    if (_values.length > _kMeasuredSampleCount) {
+      _values.removeAt(0);
     }
-
-    print(
-      'SUCCESS: Benchmark converged below ${_kNoiseThreshold * 100}%. '
-      'Noise level is ${profile.drawFrameDurationNoise * 100}%.',
-    );
-    return false;
-  }
-
-  Profile _generateProfile() {
-    final List<FrameMetrics> measuredFrames =
-        _frames.sublist(_frames.length - _kMeasuredSampleCount);
-    final Iterable<double> noiseCheckDrawFrameTimes =
-        measuredFrames.map<double>((FrameMetrics metric) =>
-            metric.drawFrameDuration.inMicroseconds.toDouble());
-    final double averageDrawFrameDurationMicros =
-        _computeMean(noiseCheckDrawFrameTimes);
-    final double standardDeviation =
-        _computeStandardDeviationForPopulation(noiseCheckDrawFrameTimes);
-    final double drawFrameDurationNoise =
-        standardDeviation / averageDrawFrameDurationMicros;
-
-    return Profile._(
-      name: name,
-      averageDrawFrameDuration:
-          Duration(microseconds: averageDrawFrameDurationMicros.toInt()),
-      drawFrameDurationNoise: drawFrameDurationNoise,
-      frames: measuredFrames,
-    );
   }
 }
 
-/// Contains metrics for a series of rendered frames.
-@immutable
+/// Base class for a profile collected from running a benchmark.
 class Profile {
-  Profile._({
-    @required this.name,
-    @required this.drawFrameDurationNoise,
-    @required this.averageDrawFrameDuration,
-    @required List<FrameMetrics> frames,
-  }) : frames = List<FrameMetrics>.unmodifiable(frames);
+  Profile({@required this.name}) : assert(name != null);
 
   /// The name of the benchmark that produced this profile.
   final String name;
 
-  /// Average amount of time [Window.onDrawFrame] took.
-  final Duration averageDrawFrameDuration;
+  /// This data will be used to display cards in the Flutter Dashboard.
+  final Map<String, Timeseries> scoreData = <String, Timeseries>{};
 
-  /// The noise, as a fraction of [averageDrawFrameDuration], measure from the [frames].
-  final double drawFrameDurationNoise;
+  /// This data isn't displayed anywhere. It's stored for completeness purposes.
```

Comment:
Technically we could build specialized displays for this data. It just won't appear in the standard score cards on the performance dashboard.